packages/ui-library/src/components/radio-group/index.ts

```diff
@@ -6,38 +6,122 @@ import { SizelessIconType } from '@boiler/icons';
-  @property() options!: RadioOption[];
   @property() hasHint = true;
   @property() groupHintMessageIcon?: SizelessIconType;
   @property() groupErrorMessage?: string;
   @property() groupHintMessage?: string;
   @property() groupErrorMessageIcon?: SizelessIconType;
   @property() legend?: string;
   @property() direction: RadioGroupDirection = 'horizontal';
-
   @property() theme: ThemeType = 'Light';
+  protected _radioElements: Element[] | undefined;
+
+  protected handleRadioClick = (event: Event) => {
+    this._radioElements?.forEach((item) => {
+      const label = item.getAttribute('label');
+      const radioInputElement = item.shadowRoot?.querySelector('input') as HTMLInputElement;
+
+      if (label === (event?.target as HTMLInputElement).id) {
+        radioInputElement.checked = true;
+      }
+    });
+
+    event.preventDefault();
+  };
+
+  firstUpdated() {
+    const slot = this.renderRoot?.querySelector('slot');
+    this._radioElements = slot?.assignedElements({ flatten: false });
+
+    setTimeout(() => {
+      {
+        this._radioElements?.forEach((item, index, elements) => {
+          if (item.localName !== RADIO_TAG_NAME) {
+            throw new Error('child component of blr-radio-group must be blr-radio');
+          }
+
+          const radioElement = <HTMLInputElement>item.shadowRoot?.querySelector('input');
+          const radioElementWrapper = <HTMLInputElement>item.shadowRoot?.querySelector('.blr-radio');
+
+          if (this.hasError) {
+            [radioElement, radioElementWrapper].forEach((element) => element.classList.add('error'));
+          }
+
+          // eslint-disable-next-line
+          // @ts-expect-error
+          if (this.sizeVariant !== item.sizeVariant) item.sizeVariant = this.sizeVariant;
+
+          radioElement.addEventListener('click', this.handleRadioClick);
+
+          Object.defineProperty(radioElement, 'checked', {
+            set: function (value) {
+              if (value === false) {
+                return;
+              }
+
+              // eslint-disable-next-line
+              // @ts-expect-error
+              item.setChecked(value);
+              elements.forEach((item) => {
+                const inputElement = item.shadowRoot?.querySelector('input') as HTMLInputElement;
+                if (inputElement.getAttribute('id') !== radioElement.getAttribute('id')) {
+                  // eslint-disable-next-line
+                  // @ts-expect-error
+                  item.setChecked(false);
+                }
+              });
+
+              this.dispatchEvent(
+                createBlrRadioGroupValueChangeEvent({
+                  originalEvent: { ...new Event('change', { bubbles: true, cancelable: true }), target: radioElement },
+                  selectedValue: radioElement.value,
+                })
+              );
+            },
+            get: function () {
+              return this.hasAttribute('checked');
+            },
+          });
```

Comment:
Logic now moved to the radio component rather than in the parent component. Changes made to the existing checked property from the developer console is not picked up by any lifecycle or lit methods so explicitly looking out for checked property is necessary.